tests/core/mountpoint/test_winfsp.py

```diff
@@ -51,6 +52,8 @@ def _entry_info_maybe_stop_loop(self, path):
             # Given this is the behavior we are testing here, make sure we
             # have actually reach this point before going further.
             mountpoint_winfsp_stop.wait()
+            # Make sure winfsp shutdown gets block by this operation
+            time.sleep(0.1)
```

Comment:
Ça me fait vraiment très peur d'avoir un test qui a besoin d'un sleep random pour fonctionner, y'a pas une autre solution ? (genre de faire du polling sur le point de montage ?)

et j'ai compris le commentaire...

Reply:
Oui ca devrait etre:

    # Make sure winfsp shutdown gets blocked by this operation

L'idée de ce test (tel que je le comprends) c'est de vérifier ce qui se passe quand `FileSystem.stop` est appelé pendant une opération. Le workflow est le suivant:
- on entre dans l'opération `entry_info` avec le chemin `/stop_loop`
- on set un événement `mountpoint_needed_stop`
- un thread reçoit l’événement et arrête le service
- le service, avant d'exécuter `FileSystem.stop()`, indique à l'opération qu'elle peut continuer son exécution

Sauf qu'à ce moment la `FileSystem.stop()` n'est pas encore appelé et en fonction du timing, l'opération peut se terminer normalement avec que le file system essaye réellement de s’arrêter. Ici, le `sleep` permet de s'assurer que la procédure de shutdown a réellement démarré avant la fin de l'opération.

N'hésite pas à me dire si tu vois une autre solution

Reply:
j'ai l'impression qu'il faudrait appeler winfsp pour savoir l'état de notre filesystème (genre `STOPPED`, `STARTING`, `READY` et `STOPPING`), avec une api comme ça ce serait trivial de faire du polling